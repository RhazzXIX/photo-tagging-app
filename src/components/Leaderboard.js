export default function Leaderboard() {
  return (
    <main>
      <button>Back to game</button>
      <button></button>
      <button></button>
      <section>
        <header>
          <p>No.</p>
          <p>Name</p>
          <p>Time</p>
        </header>
        <ul>
          <li>
            <div></div>
          </li>
        </ul>
      </section>
    </main>
  );
}
