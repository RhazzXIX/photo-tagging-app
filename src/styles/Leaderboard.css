main#leaderBoard {
  flex: auto;
  display: flex;
  align-items: center;
  flex-direction: column;
  background-color: #2b2c5a;
  gap: 12px;
  min-height: 836px;
}

main#leaderBoard > header {
  display: flex;
  justify-content: center;
  gap: 40px;
  padding: 20px;
}

main#leaderBoard > header button {
  appearance: none;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #2b2c5a;
}

main#leaderBoard > header img {
  width: 240px;
  height: auto;
  border-radius: 8px;
  box-shadow: 0 0 4px black;
}

main#leaderBoard > header img:hover {
  box-shadow: 0 0 8px grey;
}

main#leaderBoard > header img:active {
  box-shadow: 0 0 10px white;
}

main#leaderBoard > section {
  width: 560px;
  display: grid;
  grid-template-columns: 80px 1fr 1fr;
}

main#leaderBoard > section
h4:nth-of-type(3)
{
  margin-left: 14px;
}

main#leaderBoard > section ol {
  grid-column: 1 / -1;
  list-style: none;
  counter-reset: rank;
  width: 100%;
  height: 540px;
  overflow-y: auto;
}

main#leaderBoard > section li {
  margin-top: 8px;
  display: grid;
  grid-template-columns: 40px 1fr 1fr;
  justify-items: center;
  font-size: 1.6rem;
  counter-increment: rank;
}

main#leaderBoard > section li::before {
  content: counter(rank) ".)";
}
