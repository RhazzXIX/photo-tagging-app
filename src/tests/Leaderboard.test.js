import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Leaderboard from "../components/Leaderboard";

describe("Leaderboard component", () => {
  it("Renders on screen", () => {
    const { container } = render(<Leaderboard />);
    expect(container).toMatchSnapshot();
  });
});
