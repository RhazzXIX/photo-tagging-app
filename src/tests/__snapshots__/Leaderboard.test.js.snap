// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard component Renders on screen 1`] = `
<div>
  <main>
    <button>
      Back to game
    </button>
    <button />
    <button />
    <section>
      <header>
        <p>
          No.
        </p>
        <p>
          Name
        </p>
        <p>
          Time
        </p>
      </header>
      <ul>
        <li>
          <div />
        </li>
      </ul>
    </section>
  </main>
</div>
`;
